src/gd_filter.c

```diff
@@ -842,6 +842,8 @@ gaussian_coeffs(int radius, double sigmaArg) {
     double sum = 0;
     int x, n, count;
 
+    assert(radius >= 1);
```

Comment:
we should never use `assert()` in library code that users can get to.  would be nice if we could add a CI check to ban it in all library code.  tests & examples are fine to use assert/etc... of course.

we should return errors back up, and if we think it's helpful (e.g. for bad API calls), log an error via gd_error.

Reply:
This is in a static function which is only called by the `gdImageCopyGaussianBlurred()` API which alread has:

https://github.com/libgd/libgd/blob/0b8d70805d6f0b53a88902826ddc02208f8e32a1/src/gd_filter.c#L1009-L1011

This `assert()` is only a sanity check that we use the static function properly, and also a hint for static analyzers (and manual code review).

Reply:
the nuance when it's "ok" to use assert and when people are just being lazy is very hard to communicate.  people see it being used in some places and think it's ok in general.

since it's a static function, the compiler should inline it.  and the compiler should be able to see the higher level has the same check and elide one of them.  a quick check over here suggests it does, so the generated code is the same.

Reply:
enabling extra debug in a library should not cause it to abort and kill a process.  assert usage in programs is different from libraries.  we should have no asserts in the gd library regardless of its compilation mode.  the existing asserts are "ok" in the sense that they always force NDEBUG which means the calls are empty.  interesting that coverity accepts that behavior ... I can see both sides there.